libs/shared/base-abstract-classes/src/service/editors/multi-row-editor.service.ts

```diff
@@ -187,16 +194,16 @@ export abstract class MultiRowEditorService<T extends TableRow> extends EditorSe
     this.updateFullQuery();
   }
 
-  protected addIdToNewRow(newRow): void {
-    newRow[this._entityIdField] = Number.parseInt(this.loadedEntityId, 10);
+  protected addIdToNewRow(newRow: T): void {
+    newRow[this._entityIdField as keyof T] = Number.parseInt(this.loadedEntityId, 10) as T[keyof T];
   }
 
-  addNewRow(): void {
-    const newRow = new this._entityClass();
+  addNewRow(copySelectedRow = false): void {
+    const newRow: T = copySelectedRow && this.hasSelectedRow() ? { ...this.getSelectedRow() } : new this._entityClass();
```

Comment:
Note for reviewer: this is the actual copy feature

Reply:
not sure if we have to replace `{ ...this.getSelectedRow() }` with a deep clone of the object using `structuredClone(this.getSelectedRow())`

Reply:
@Helias why a deep clone? all the properties of each row are always strings or numbers, never objects.

Reply:
then it's ok